### ISSUE TITLE
[Fluent design] Current UpdateWithTags interface requires PUT call for cleaning all Tags on the resource

### ISSUE THREAD

**Issue by hovsepm** [Azure/azure-sdk-for-java#1069]
Changing update to Patch will reduce network payload and it would be good to add withoutTags() method that will clean underlying map and send empty collection to Azure.

**Comment by praries880** [comment]
@hovsepm does this issue belong in https://github.com/Azure/azure-libraries-for-java?

**Comment by hovsepm** [comment]
@praries880 yeah should go to libraries repo. We used to have Fluent SDK in this repo initially and then moved to a separate repo.
